File: backend/wsp/src/main/java/wsp/entity/ShoppingItem.java
package wsp.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

/**
 * Encja produktu znajdującego się na grupowej liście zakupów.
 */
@Entity
@Table(name = "shopping_items")
@Getter
@Setter
public class ShoppingItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String quantity;

    @Column(nullable = false)
    private boolean bought = false;

    @ManyToOne(optional = false)
    @JoinColumn(name = "group_id", nullable = false)
    private UserGroup group;
}

File: client/wsp/lib/features/shopping/models/shopping_item.dart
class ShoppingItem {
  const ShoppingItem({
    required this.id,
    required this.groupId,
    required this.name,
    required this.quantity,
    required this.bought,
  });

  final int id;
  final int groupId;
  final String name;
  final String quantity;
  final bool bought;

  factory ShoppingItem.fromJson(Map<String, dynamic> json) {
    return ShoppingItem(
      id: json['id'] as int,
      groupId: json['groupId'] as int,
      name: json['name'] as String,
      quantity: json['quantity'] as String,
      bought: json['bought'] as bool,
    );
  }
}
